plugins/out_http/http.c

```diff
@@ -328,13 +359,179 @@ static int http_gelf(struct flb_out_http *ctx,
+    mpack_tag_t mtag;
+    size_t key_count;
+    size_t i;
+    size_t len;
+    char **headers = NULL;
+    size_t str_count;
+
+    mtag = mpack_read_tag(reader);
+    if (mtag.type != mpack_type_map) {
+        goto err;
+    }
+
+    key_count = mpack_tag_map_count(&mtag);
+    if (!key_count) {
+        goto err;
+    }
+
+
+    str_count = key_count * 2 + 1;
+    headers = flb_calloc(str_count, sizeof *headers);
+
+    for (i = 0; i < key_count; i++) {
+        mtag = mpack_read_tag(reader);
+        if (mtag.type != mpack_type_str && mtag.type != mpack_type_bin) {
+            goto err;
+        }
+
+        len = mpack_tag_bytes(&mtag);
+        headers[i * 2] = strndup(reader->data, len);
+
+        if (!headers[i]) {
+            goto err;
+        }
+
+        reader->data += len;
+        mtag = mpack_read_tag(reader);
+        if (mtag.type != mpack_type_str && mtag.type != mpack_type_bin) {
+            goto err;
+        }
+
+        len = mpack_tag_bytes(&mtag);
+        headers[i * 2 + 1] = strndup(reader->data, len);
+
+        if (!headers[i]) {
+            goto err;
+        }
+
+        reader->data += len;
+    }
+
+    return headers;
+
+err:
+    if (headers) {
+        for (i = 0; i < str_count; i++) {
+            if (headers[i]) {
+                flb_free(headers[i]);
+            }
+        }
+        flb_free(headers);
+    }
+    return NULL;
+}
+
+static int post_all_requests(struct flb_out_http *ctx,
+                             const char *data, size_t size,
+                             flb_sds_t body_key,
+                             flb_sds_t headers_key,
+                             struct flb_event_chunk *event_chunk)
+{
+    int ret;
+    struct flb_time t;
+    mpack_tag_t mtag;
+    mpack_reader_t reader;
+    size_t key_count;
+    size_t i;
+    size_t len;
+    const char *body;
+    size_t body_size;
+    bool body_found;
+    bool is_body;
+    bool headers_found;
+    bool is_headers;
+    char **headers;
+    const char *record_start;
+    size_t record_size;
+    int record_count = 0;
+
+    mpack_reader_init_data(&reader, data, size);
+
+    while (size > 0) {
+        headers = NULL;
+        body_found = false;
+        headers_found = false;
+        record_start = reader.data;
+        record_size = 0;
+
+        ret = flb_time_pop_from_mpack(&t, &reader);
+        if (ret) {
+            return ret;
+        }
+
+        mtag = mpack_read_tag(&reader);
+        if (mtag.type != mpack_type_map) {
+            return -1;
+        }
+
+        key_count = mpack_tag_map_count(&mtag);
+
+        for (i = 0; i < key_count; i++) {
+            is_body = false;
+            is_headers = false;
+            mtag = mpack_read_tag(&reader);
+            if (mtag.type != mpack_type_str && mtag.type != mpack_type_bin) {
+                return -1;
+            }
+
+            len = mpack_tag_bytes(&mtag);
+            is_body = len == flb_sds_len(body_key) &&
```

Comment:
please use an `if` statement for this:

```c
if (len == flb_sds_len(...) && ...) {
    is_body = FLB_TRUE;
}
else {
..
}
```

same for the other variables below
```

Reply:
I've removed this code due to the mpack -> msgpack-c change